src/transformers/models/albert/tokenization_albert_fast.py

```diff
@@ -135,8 +134,6 @@ def __init__(
         mask_token="[MASK]",
         **kwargs
     ):
```

Comment:
Kindly check out the following code:

https://github.com/huggingface/transformers/blob/c28bc80bbbe0cf5cb289944ec2374d5397ff6e31/src/transformers/tokenization_utils_fast.py#L104-L123

There are four ways to instantiate FastTokenizer. The first two are not applicable to this case as they are either instantiated from an existing a fast tokenizer object or a fast tokenizer file i.e. The `[MASK]` token handled or not by `AlbertTokenizerFast.__init__` doesn't matter because all states have already stored in the object or file (explained below). 
The last two are instantiated from a converted slow tokenizer which handles everything (including the `[MASK]` token) in its constructor then is converted to the fast one. 

So if we start from 

`tokenizer = AlbertTokenizerFast("tests/fixtures/spiece.model")`

The running flow will be

`AlbertTokenizerFast.__init__`
`PreTrainedTokenizerFast.__init__`
`self.slow_tokenizer_class is not None (fourth condition)` 
`slow_tokenizer = self.slow_tokenizer_class(*args, **kwargs)` << Pass args & kwargs to slow tokenizer.
`fast_tokenizer = convert_slow_tokenizer(slow_tokenizer)`
`kwargs.update(slow_tokenizer.init_kwargs)` << Update FastTokenizer's kwargs from the slow one
`super().__init__(**kwargs)`
...

And, if we start from pretrained instead

`tokenizer = AlbertTokenizerFast.from_pretrained("albert-base-v2")`

The running flow will be

`AlbertTokenizerFast.__init__`
`PreTrainedTokenizerFast.__init__`
`fast_tokenizer_file is not None and not from_slow (second condition)`
`fast_tokenizer = TokenizerFast.from_file(fast_tokenizer_file)`
`super().__init__(**kwargs)`  <<The config in kwargs doesn't affect its behavior as the fast tokenizer was already created from the file.

So there would result in another weird behavior. That is, even though I've set `cls_token` and `bos_token` to `[MASK]`, the tokenized outputs ramain unchanged:

```
tokenizer = AlbertTokenizerFast.from_pretrained("albert-base-v2", cls_token="[MASK]", bos_token="[MASK]")
tokenized_ids = tokenizer("hello").input_ids

print(tokenizer.cls_token) # '[MASK]'
print(tokenizer.convert_ids_to_tokens(tokenized_ids)) # ['[CLS]', '▁hello', '[SEP]']
```

Reply:
@qqaatw, thanks again for your detailed and precise answer! :smile: 
 
Indeed, I went a bit fast by ignoring the case where the fast tokenizer is automatically built from the slow: you are completely right! 

Nevertheless, we try to support - as much as possible - the initialization from a `tokenizer` object or a `tokenizer.json` file even if the tokenizer of the "reference" checkpoints on the Hub do not use this method. And for these last two cases, it seems to me that we need to keep the line . :blush:

Reply:
> So there would result in another weird behavior. That is, even though I've set cls_token and bos_token to [MASK], the tokenized outputs ramain unchanged:

Indeed, to actually change the special tokens added to the sequence it would be necessary to modify the tokenizer wrapped by the fast tokenizer and stored in the attribute `backend_tokenizer`. In this case, it would be necessary to modify the `PostProcessor` of the `backend_tokenizer` and for the moment it seems to me that the idea of the initialization of the fast tokenizers was rather to give to the wrapper the information already contained in the `backend_tokenizer`.

The addition of `self.sanitize_special_tokens()` is indeed diverging from this design and that's why I'm still not 100% that it's the best way to solve the problem you raised (the fact that the reloaded tokenizer after a registration is not the same anymore).

Reply:
I understood your point. I've added the line back to make it more compatible to every use-case.
And I also removed the change of adding `self.sanitize_special_tokens()` in `PreTrainedTokenizerFast.__init__` to keep this PR focusing on solving mask token problem of **Albert**.

If you finally decide to adjust the workflow i.e. always sanitizing special tokens even if the special tokens do not exist in the vocab, I'm happy to open another PR for it.